plugins/draw-tools.js

```diff
@@ -659,7 +659,7 @@ window.plugin.drawTools.snapToPortals = function () {
     if (minGuid) {
       var pll = window.portals[minGuid].getLatLng();
       if (pll.lat !== latlng.lat || pll.lng !== latlng.lng) {
-        return L.latLng(pll);
+        return new L.LatLng(pll.lat, pll.lng);
```

Comment:
Interesting... can you pls explain how is this fixing the issue?
Those two lines should have the same effect... from the first look 🤷‍♂️

Reply:
Ah... we need to make a copy...

Anyways, tested on my S25 and it works with and without this fix 🤷‍♂️

Reply:
L.latLng is a shortcut for L.toLatLng. which is used to convert a bunch of formats to a LatLng object.
But if the parameter is already a LatLng object it simply return that object.


`a = L.latLng(1,1);
b= L latLng(a);
b lat =2;

console.log(a);
// { lat: 2, lng: 1} 
// a and b are the same object reference`

In our case instead of setting the coordinates, snap sets the location reference to the portal marker location object.
So when the drawtool marker is moved it also alters the portal marker position.